lib/msf/core/rhosts_walker.rb

```diff
@@ -146,7 +146,6 @@ def parse(value, datastore)
             Rex::Socket::RangeWalker.new(value).each_host do |rhost|
               overrides = {}
               overrides['UNPARSED_RHOSTS'] = value
-              overrides['RHOSTS'] = rhost[:address]
```

Comment:
IIRC the reason why this was done this way was so that when a hostname resolves to multiple IP addresses, the RangeWalker is supposed handle them all. So if example.com is on 1.1.1.1 and 2.2.2.2 then we should iterate twice with RHOSTS always set to the IP address (once for each) and the VHOST set to the name of example.com to ensure that both HTTP servers are tested.

By removing that, I think what's going to happen is we may not be scanning both targets explicitly, it would depend on whatever IP address the socket connects to because it's using the name which is open for interpretation vs the IP address that is very clear about where the connection goes.